### PR TITLE
feat(behavior_path _planner): divide planner manager modules into dependent slots

### PR DESCRIPTION
**Pull request by soblin**
## Description

divide behavior_path_planner manager into slots

## Related links

https://github.com/autowarefoundation/autoware_launch/pull/1091

**Parent Issue:**

- https://tier4.atlassian.net/browse/RT1-6875

## How was this PR tested?

https://evaluation.tier4.jp/evaluation/reports/2114008f-ad21-5669-aac0-4872be67d50f?project_id=prd_jt

### avoidance + lane_change + goal_planner + dynamic_avoidance

https://github.com/user-attachments/assets/2424763e-0d92-4220-a720-ec2ec1036cee

### side_shift + modified_goal_planner + dynamic_avoidance

https://github.com/user-attachments/assets/567f9ecc-7c45-49ff-8d53-774fcb21c83a

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by kyoichi-sugahara** [comment]
@soblin 
I think that since it significantly deviates from the current documentation explaining the architecture of the behavior_path_planner, we need to note somewhere in the documentation that the current explanation is deprecated and that modifications are necessary to align with the current implementation. What do you think?

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

**Review by yuki-takagi-66 — APPROVED** [review]
LGTM about dynamic_obstacle_avoidance

**Review by takayuki5168 — APPROVED** [review]
LGTM for dynamic obstacle avoidance

File: planning/behavior_path_planner/autoware_behavior_path_planner/src/planner_manager.cpp [hunk 1]
```diff
@@ -674,149 +732,116 @@ std::pair<SceneModulePtr, BehaviorModuleOutput> PlannerManager::runRequestModule
...
 
   /**
    * remove failure modules. these modules' outputs are discarded as invalid plan.
    */
-  {
-    const auto itr = std::find_if(
-      approved_module_ptrs_.begin(), approved_module_ptrs_.end(),
-      [](const auto & m) { return m->getCurrentStatus() == ModuleStatus::FAILURE; });
-    if (itr != approved_module_ptrs_.end()) {
-      deleted_modules.push_back(*itr);
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_planner/src/planner_manager.cpp:795]: previous version

File: planning/behavior_path_planner/autoware_behavior_path_planner/src/planner_manager.cpp [hunk 2]
```diff
@@ -828,199 +853,124 @@ BehaviorModuleOutput PlannerManager::runApprovedModules(
...
-    const auto itr =
-      std::find_if(approved_module_ptrs_.begin(), approved_module_ptrs_.end(), success_module_cond);
-
-    if (std::any_of(itr, approved_module_ptrs_.end(), [](const auto & m) {
-          return m->isCurrentRouteLaneletToBeReset();
-        }))
-      resetCurrentRouteLanelet(data);
-
-    std::copy_if(
-      itr, approved_module_ptrs_.end(), std::back_inserter(deleted_modules), success_module_cond);
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_planner/src/planner_manager.cpp:869]: previous version

**Pull request merged**